fpm/src/fpm.f90

```diff
@@ -154,8 +154,49 @@ subroutine build_model(model, settings, package, error)
     end do
     if (allocated(error)) return
 
+    ! Check for duplicate modules
+    call check_modules(model)
 end subroutine build_model
 
+! Check for duplicate modules
+subroutine check_modules(model) 
```

Comment:
I suggest a slightly more descriptive `check_modules_for_duplicates`. It's rather long, but it's called only once so I think a descriptive name is better.